src/options_app.c

```diff
@@ -0,0 +1,1479 @@
+            CopyWindowToVram(w1);
+            OptionsApp_PrintTextFrameString(data, data->frameNumText, TRUE);
+            data->unk10_21 = TRUE;
+            return;
+        case MENU_ENTRY_SOUND_METHOD:
+            GF_SndSetMonoFlag(data->menuEntries[menuEntryId].value);
+            break;
+        case MENU_ENTRY_BUTTON_MODE:
+            Options_SetButtonModeOnMain(NULL, data->menuEntries[menuEntryId].value);
+            break;
+        case MENU_ENTRY_TEXT_SPEED:
+            Options_SetTextSpeed(data->playerOptions, data->menuEntries[menuEntryId].value);
+            OptionsApp_PrintTextFrameString(data, data->frameNumText, FALSE);
+            break;
+    }
+
+    // if (data->menuEntries[menuEntryId].numStrings > 0)
+    for (u16 i = 0; i < data->menuEntries[menuEntryId].numStrings; i++) {
+        u32 color = (i == data->menuEntries[menuEntryId].value) ? MAKE_TEXT_COLOR(1, 2, 0) : MAKE_TEXT_COLOR(15, 2, 0);
+        u16 x = ov54_021E6D00[menuEntryId][i] - (FontID_String_GetWidth(0, data->menuEntries[menuEntryId].strings[i], 0) / 2);
+        AddTextPrinterParameterizedWithColor(w1, 0, data->menuEntries[menuEntryId].strings[i], x, y, TEXT_SPEED_NOTRANSFER, color, NULL);
+    }
+
+    CopyWindowToVram(w1);
+}
+#else
+static asm void ov54_021E6418(OptionsApp_Data *data, u16 menuEntryId) {
+    push {r3, r4, r5, r6, r7, lr}
+    sub sp, #0x20
+    add r6, r1, #0
+    str r0, [sp, #0x10]
+    ldr r2, [pc, #0x19C] // _021E65C0 ; =_021E6C30
+    mov r1, #0x18
+    add r0, r6, #0
+    mul r0, r1
+    add r0, r0, #5
+    str r0, [sp, #0x14]
+    ldrsb r2, [r2, r6]
+    lsl r0, r1, #4
+    str r0, [sp]
+    ldr r3, [sp, #0x14]
+    ldr r0, [sp, #0x10]
+    add r2, #0x6c
+    lsl r2, r2, #0x10
+    lsl r3, r3, #0x10
+    str r1, [sp, #4]
+    add r0, #0x44
+    mov r1, #0
+    lsr r2, r2, #0x10
+    lsr r3, r3, #0x10
+    bl FillWindowPixelRect
+    cmp r6, #5
+    bhi _021E652E
+    add r0, r6, r6
+    add r0, pc
+    ldrh r0, [r0, #6]
+    lsl r0, r0, #0x10
+    asr r0, r0, #0x10
+    add pc, r0
+_021E645A: // jump table
+    DCD 0x00D200AC // _021E6508 - _021E645A - 2 ; case 0
+                   // _021E652E - _021E645A - 2 ; case 1
+    DCD 0x008200D2 // _021E652E - _021E645A - 2 ; case 2
+                   // _021E64DE - _021E645A - 2 ; case 3
+    DCD 0x000A0096 // _021E64F2 - _021E645A - 2 ; case 4
+                   // _021E6466 - _021E645A - 2 ; case 5
+_021E6466:
+    ldr r4, [sp, #0x10]
+    mov r0, #0x54
+    add r7, r6, #0
+    mul r7, r0
+    add r4, #0x86
+    ldr r0, [sp, #0x10]
+    ldrh r1, [r4, r7]
+    add r0, #0x88
+    add r5, r0, r7
+    lsl r1, r1, #2
+    mov r0, #0
+    ldr r1, [r5, r1]
+    add r2, r0, #0
+    bl FontID_String_GetWidth
+    mov r1, #6
+    add r2, r6, #0
+    mul r2, r1
+    ldr r1, [pc, #0x138] // _021E65C4 ; =ov54_021E6D00
```

Comment:
you can probably just put the = directly, and the compiler *should* just generate the data correctly at the end